src/adapters/aave-v2/products/pool/aaveV2BasePoolAdapter.ts

```diff
@@ -0,0 +1,361 @@
+      eventResults,
+    })
+  }
+
+  async getClaimedRewards(_input: GetEventsInput): Promise<MovementsByBlock[]> {
+    return []
+  }
+
+  async getOneDayProfit({
+    userAddress,
+    blockNumber,
+  }: GetProfitsInput): Promise<ProfitsTokensWithRange> {
+    const toBlock = blockNumber
+    const fromBlock = toBlock - AVERAGE_BLOCKS_PER_DAY[this.chainId]
+
+    const [currentValues, previousValues] = await Promise.all([
+      this.getPositions({
+        userAddress,
+        blockNumber: toBlock,
+      }).then(formatProtocolTokenArrayToMap),
+      this.getPositions({
+        userAddress,
+        blockNumber: fromBlock,
+      }).then(formatProtocolTokenArrayToMap),
+    ])
+
+    const tokens = await Promise.all(
+      Object.values(currentValues).map(
+        async ({ protocolTokenMetadata, underlyingTokenPositions }) => {
+          const getEventsInput: GetEventsInput = {
+            userAddress,
+            protocolTokenAddress: protocolTokenMetadata.address,
+            fromBlock,
+            toBlock,
+          }
+
+          const [withdrawals, deposits] = await Promise.all([
+            this.getWithdrawals(getEventsInput).then(aggregateTrades),
+            this.getDeposits(getEventsInput).then(aggregateTrades),
+          ])
+
+          const profits = calculateProfit({
+            deposits,
+            withdrawals,
+            currentValues: underlyingTokenPositions,
+            previousVales:
+              previousValues[protocolTokenMetadata.address]
+                ?.underlyingTokenPositions ?? {},
+          })
+
+          return {
+            ...protocolTokenMetadata,
+            type: TokenType.Protocol,
+            tokens: Object.values(underlyingTokenPositions).map(
+              (underlyingToken) => {
+                return {
+                  ...underlyingToken,
+                  profitRaw: profits[underlyingToken.address]!,
+                  profit: formatUnits(
+                    profits[underlyingToken.address]!,
+                    underlyingToken.decimals,
+                  ),
+                  type: TokenType.Underlying,
+                }
+              },
+            ),
+          }
+        },
+      ),
+    )
+
+    return { tokens, fromBlock, toBlock }
+  }
+
+  async getPricePerShare({
+    protocolTokenAddress,
+  }: GetPricesInput): Promise<ProtocolPricePerShareToken> {
+    const { protocolToken, underlyingToken } =
+      this.fetchProtocolTokenMetadata(protocolTokenAddress)
+
+    const pricePerShareRaw = BigInt(1 * 10 ** protocolToken.decimals)
+
+    const pricePerShare = formatUnits(
+      pricePerShareRaw,
+      underlyingToken.decimals,
+    )
+
+    return {
+      ...protocolToken,
+      share: 1,
+      type: TokenType.Protocol,
+      tokens: [
+        {
+          ...underlyingToken,
+          type: TokenType.Underlying,
+          pricePerShareRaw,
+          pricePerShare,
+        },
+      ],
+    }
+  }
+
+  async getPositions({
```

Comment:
Again, very similar code to the one used at stargate. The only difference is the balance calculation for the underlying token.